lib/src/http.rs

```diff
@@ -1449,11 +1450,16 @@ impl Proxy {
         self.listeners.get(token).map(Clone::clone)
     }
 
-    pub fn remove_listener(&mut self, address: SocketAddr) -> bool {
+    pub fn remove_listener(&mut self, remove: RemoveListener) -> anyhow::Result<()> {
         let len = self.listeners.len();
+        self.listeners
+            .retain(|_, l| l.borrow().address != remove.address);
 
-        self.listeners.retain(|_, l| l.borrow().address != address);
-        self.listeners.len() < len
+        if self.listeners.len() < len {
+            Ok(())
+        } else {
+            bail!("no HTTP listener to remove at address {:?}", remove.address)
```

Comment:
Not sure about this. To me, we should not return an error when the listener is not found

Reply:
was thinking the same. How about we return Ok() and juste log the line?

Reply:
We could opposite the condition, log something if the condistion is true and always return `Ok(())`

Reply:
that's what I did: if there is no listener to remove, log it, and return Ok(()) anyway